scripts/deploy.js

```diff
@@ -89,14 +126,30 @@ async function deployAll(
 
     const interchainTokenServiceAddress = await getCreate3Address(create3Deployer.address, wallet, deploymentKey);
     const tokenManagerDeployer = await deployContract(wallet, 'TokenManagerDeployer', []);
-    const interchainToken = await deployContract(wallet, 'InterchainToken', [interchainTokenServiceAddress]);
-    const interchainTokenDeployer = await deployContract(wallet, 'InterchainTokenDeployer', [interchainToken.address]);
+
+    // Determine if this is a Hyperliquid chain
+    const isHyperliquidChain = chainName.toLowerCase().includes('hyperliquid');
+
+    // Deploy only the appropriate token implementation based on chain type
+    let interchainToken, hyperliquidInterchainToken, interchainTokenDeployer;
+
+    if (isHyperliquidChain) {
+        hyperliquidInterchainToken = await deployContract(wallet, 'HyperliquidInterchainToken', [interchainTokenServiceAddress]);
+        interchainTokenDeployer = await deployContract(wallet, 'InterchainTokenDeployer', [hyperliquidInterchainToken.address]);
+    } else {
+        interchainToken = await deployContract(wallet, 'InterchainToken', [interchainTokenServiceAddress]);
+        interchainTokenDeployer = await deployContract(wallet, 'InterchainTokenDeployer', [interchainToken.address]);
+    }
```

Comment:
```suggestion
    const interchainToken = await deployContract(wallet, (isHyperliquidChain) ? 'HyperliquidInterchainToken' : 'InterchainToken', [interchainTokenServiceAddress]);
    const interchainTokenDeployer = await deployContract(wallet, 'InterchainTokenDeployer', [interchainToken.address]);
```
use the ternary operator as I previously suggested